src/System.Windows.Forms/src/System/Windows/Forms/FolderBrowserDialog.cs

```diff
@@ -228,10 +270,24 @@ private void SetDialogProperties(IFileDialog dialog)
 
             dialog.SetOptions(FOS.PICKFOLDERS | FOS.FORCEFILESYSTEM | FOS.FILEMUSTEXIST);
 
+            if (!string.IsNullOrEmpty(InitialDirectory))
+            {
+                try
+                {
+                    IShellItem initialDirectory = GetShellItemForPath(InitialDirectory);
+
+                    dialog.SetDefaultFolder(initialDirectory);
+                    dialog.SetFolder(initialDirectory);
+                }
+                catch (FileNotFoundException)
+                {
+                }
+            }
+
             if (!string.IsNullOrEmpty(_selectedPath))
             {
                 string parent = Path.GetDirectoryName(_selectedPath);
-                if (parent is null || !Directory.Exists(parent))
+                if (parent is null || !Directory.Exists(parent) || !string.IsNullOrEmpty(InitialDirectory))
```

Comment:
This should be a little more performant
```suggestion
                if (parent is null || !string.IsNullOrEmpty(InitialDirectory) || !Directory.Exists(parent))
```